code/obj/item/gun/kinetic.dm

```diff
@@ -2144,6 +2144,136 @@ ABSTRACT_TYPE(/obj/item/survival_rifle_barrel)
+
+	examine()
+		. = ..()
+		. += "\n \n It is set to fire " //differentiate the FLW examine text from the default gun examine text
+		switch(firemode)
+			if(ONE_BARREL)
+				. += "one barrel."
+			if(TWO_BARRELS)
+				. += "two barrels."
+			if(ALL_BARRELS)
+				. += "all four barrels!"
+		if(guaranteed_uses <= 0)
+			. += " It seems severely damaged!"
+		else if (guaranteed_uses < 6)
+			. += " It seems damaged."
+
+	shoot(var/target, var/start, var/mob/user)
+
+		var/i //Because we can't initialize variables when we're declaring the loop. thanks, DM.
+		var/priorarmmo = src.ammo.amount_left
+
+		//Go up to the limit defined by the firing mode, but don't exceed however many bullets are in the gun BEFORE we started firing
+		for(i=1, ((i <= priorarmmo) && (i <= firemode)), i++)
+			..() //shoot an additional ith time
+			guaranteed_uses-- //damage the gun an additional ith time
+
+		//you're shooting multiple shotgun shells out of a garbage gun at the same time. don't think there won't be consequences
+		if((firemode == TWO_BARRELS) && (priorarmmo != 0))
+			boutput(user, SPAN_ALERT("The [src] jumps in your hands!"))
+			animate_shake(user, 2, 3, 3, 0, 0)
+			user.do_disorient(stamina_damage = 20, knockdown = 0, stunned = 0, disorient = 5, remove_stamina_below_zero = 0)
+		if((firemode == ALL_BARRELS) && (priorarmmo != 0))
+			sleep(0.3) //give it a micro-delay
+			if (src.canshoot(user))
+				boutput(user, SPAN_ALERT("The [src] kicks like a damn mule!"))
+				animate_shake(user, 2, 5, 5, 0, 0)
+				//this might seem punishing but keep in mind this is still FOUR whole shotgun shells at once.
+				user.do_disorient(stamina_damage = 40, knockdown = 0, stunned = 0, disorient = 20, remove_stamina_below_zero = 0)
+
+		//check the gun's condition, break as needed
+		if(guaranteed_uses < 0)
+			//warn the user that they're in the danger zone
+			playsound(src.loc, 'sound/impact_sounds/Generic_Snap_1.ogg', 50, 1)
+			user.visible_message(SPAN_COMBAT("The [src] [pick(list("rattles!", "bulges!", "thunks!", "makes a concerning click...", "cracks!"))]"))
+			if (prob(5 + (guaranteed_uses*-5))) //roll for failure. Since [uses] is now negative, we need another minus sign to cancel out
+
+				user.visible_message(SPAN_ALERT("[user]'s [src] makes a severe-sounding bang!"), SPAN_ALERT("The [src] gives out!"))
+
+				if(firemode == ALL_BARRELS) //ohhh, you REALLY fucked up now.
+					explosion(src, get_turf(src), 0, 0.5, 1.5, 4)
+
+				//replace the gun with broken version
+				var/obj/item/brokenquadbarrel/broken = new /obj/item/brokenquadbarrel
+				user.drop_item(src)
+				user.put_in_hand_or_drop(broken)
+				qdel(src)
+
+	attackby(obj/item/I, mob/user)
+		if (isweldingtool(I)) //are we repairing?
+			var/obj/item/weldingtool/welder = I
+			if(guaranteed_uses >= MAX_USES)
+				boutput(user, SPAN_NOTICE("The [src] doesn't seem to be all that damaged."))
+			else if (welder.try_weld(user, 2, 2))
```

Comment:
> helping to reinforce the hit-and-run salvager gameplay loop

As implemented this seems to only be gated by the amount of welder fuel you have.  Just toss it in a backpack and spam your welder on it, and get it back out again.  I think I'd rather see this as an actionbar requiring full disengagement before being used again.